### ISSUE TITLE
No temperature in UI, CentOS 7 

### ISSUE THREAD

**Issue by bellwood** [pi-hole/AdminLTE#771]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/AdminLTE/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/AdminLTE/issues)).

**How familiar are you with the the source code relevant to this issue?:**

1

---
**Expected behaviour:**

`Temperature should be picked up, however temp1_input does not exist and I do not have a thermal_zone0`

**Actual behaviour:**

```
[root@pihole hwmon0]# pwd
/sys/class/hwmon/hwmon0
[root@pihole hwmon0]# ls -al
lrwxrwxrwx 1 root root    0 Jun 13 21:37 device -> ../../../coretemp.0
-r--r--r-- 1 root root 4096 Jun 13 21:37 name
drwxr-xr-x 2 root root    0 Jun 13 21:37 power
lrwxrwxrwx 1 root root    0 Jun 13 21:37 subsystem -> ../../../../../class/hwmon
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp2_crit
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp2_crit_alarm
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp2_input
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp2_label
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp3_crit
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp3_crit_alarm
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp3_input
-r--r--r-- 1 root root 4096 Jun 13 21:37 temp3_label
-rw-r--r-- 1 root root 4096 Jun 13 21:37 uevent
[root@pihole hwmon0]# cd /sys/class/thermal/thermal_zone0/
-bash: cd: /sys/class/thermal/thermal_zone0/: No such file or directory
[root@pihole hwmon0]# cd /sys/class/thermal
[root@pihole thermal]# ls -al
lrwxrwxrwx  1 root root 0 May 31 21:15 cooling_device0 -> ../../devices/virtual/thermal/cooling_device0
lrwxrwxrwx  1 root root 0 May 31 21:15 cooling_device1 -> ../../devices/virtual/thermal/cooling_device1
lrwxrwxrwx  1 root root 0 May 31 21:15 cooling_device2 -> ../../devices/virtual/thermal/cooling_device2
lrwxrwxrwx  1 root root 0 May 31 21:15 cooling_device3 -> ../../devices/virtual/thermal/cooling_device3
```

**Steps to reproduce:**

`See above`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`6287x19029`

**Troubleshooting undertaken, and/or other relevant information:**

* CentOS 7 minimal install.
* Kernel: 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Unable to symlink temp2_input to temp1_input (did read previous discord thread about this)

Possible to make this a config option to direct the service where to locate the sensor?

Edit: spelling

**Comment by bellwood** [comment]
`sensors -u` does spit out a temp1_input:

```
...
temp1:
  temp1_input: 30.000
  temp1_max: 56.000
  temp1_max_hyst: -106.000
  temp1_alarm: 0.000
  temp1_type: 4.000
  temp1_beep: 0.000
...
```

...and that is located:

```
[root@pihole sys]# find . -name "temp1_input"
./devices/platform/w83627hf.656/temp1_input
```

**Comment by bellwood** [comment]
```
[root@pihole sys]# ls /sys/class/hwmon/hwmon*/temp*_*
/sys/class/hwmon/hwmon0/temp2_crit        /sys/class/hwmon/hwmon0/temp2_input  /sys/class/hwmon/hwmon0/temp3_crit        /sys/class/hwmon/hwmon0/temp3_input
/sys/class/hwmon/hwmon0/temp2_crit_alarm  /sys/class/hwmon/hwmon0/temp2_label  /sys/class/hwmon/hwmon0/temp3_crit_alarm  /sys/class/hwmon/hwmon0/temp3_label
[root@pihole sys]# cat /sys/class/hwmon/hwmon0/temp3_label
Core 1
[root@pihole sys]# cat /sys/class/hwmon/hwmon0/temp3_input
4000
[root@pihole sys]# cat /sys/class/hwmon/hwmon0/temp2_label
Core 0
[root@pihole sys]# cat /sys/class/hwmon/hwmon0/temp2_input
0
```

So for whatever reason, the kernel is starting its numbering at 2 instead of 1

**Comment by bellwood** [comment]
My work around:

###### /etc/pihole/setupVars.conf
```
TEMP_SENSOR=/sys/class/hwmon/hwmon0/temp3_input
```

###### header.php
```
    if(file_exists($setupVars['TEMP_SENSOR']))
    {
     	$output = rtrim(file_get_contents($setupVars['TEMP_SENSOR']));
    }
```
(changing the original first `if` to an `elseif`)

If I get can get an intuitive listing of the thermal inputs into the settings pages for the user to choose from I will make a PR

**Comment by rrobgill** [comment]
There are many possible locations for the temperature sensors, some systems will have multiple, even many sensors present, and the names are not always so intuitive. Adding them all is essentially impossible.

My understanding is that temperature reporting in the web interface is likely to be discontinued in a future release, as it is not a function of providing an ad blocking DNS service.

If you wish to monitor the temperature of a remote system, [lm-sensors](https://github.com/groeck/lm-sensors), which provides the sensor command you quoted, does have ability to export this information via SNMP.

**Comment by thezerocool** [comment]
I had a similar issue.  Temp file /sys/class/thermal/thermal_zone0/temp exists but gave a value of 1 meaning that it was filtered later in the headers.php file.  Having the file check as successful is not enough of a check on the nested IF, can I suggest it also needs to check for a valid value as well.  Without checking the value the nested IF didnt test other files or hit the ELSE statement.  No temperature was visible.

I am running a raspberry pi 1B running TCL - tiny core linux with the latest pihole as of last week.

My workaround is stinky as it requires a permission mod, its also usercase specific but I will share anyway for consideration of a TCL appropriate solution...

I allowed www-data permissions for the video group to allow it access to a exec() bash command and got the temperature that way.  `sudo usermod -aG video www-data`  Reboot required to take effect.

I replaced lines 21 to 32 in header.php with the following code...

```
exec("vcgencmd measure_temp",$strData,$o);
$strData = preg_replace("([[:punct:]]|[[:alpha:]]| )",'',$strData[0]);
$intData = (int)$strData/10;   // measured in milli-degrees later
$output = $intData;
```
